heron/proto/ckptmgr.proto

```diff
@@ -0,0 +1,270 @@
+package heron.proto.ckptmgr;
+option java_package = "com.twitter.heron.proto.ckptmgr";
+option java_outer_classname = "CheckpointManager";
+
+import "common.proto";
+import "physical_plan.proto";
+
+// The messages in this file are for stateful processing and providing
+// exactly once semantics. Below are the various sequence of events that
+// happen in a few(non exhaustive) scenarios
+//
+// On Startup
+//   For topologies needing exact once, the startup sequence of
+// tmaster/stmgr/instance is a little different. Normally, upon
+// all stmgrs connecting to tmaster, tmaster computes pplan
+// and distributes to stmgrs, which inturn send it to all their
+// local instances. As soon as instances get their pplan, they can
+// start processing(spouts emitting tuples, etc). However for
+// exactly once semantics, this is a little different. First
+// in addition to the pplan, tmaster needs to send the last
+// consistent checkpoint id(if any) to recover from as well.
+// Secondly because instances are all busy recovering their state,
```

Comment:
"because instances are all busy recovering their state" assumes that the reader knows the startup sequence of events, which is what we're trying to describe. Could you list the sequence of startup events for stateful topologies? It should be very clear to the reader the order of things like fetching the packing plan, initializing state, registering with stream managers, physical plan generation, etc.